common/openapi.yml

```diff
@@ -267,6 +286,59 @@ components:
         catalogEntry:
           type: string
 
+    DatasetPreviewTablesResponse:
+      type: object
+      properties:
+        tables:
+          type: array
+          items:
+            $ref: '#/components/schemas/TableModel'
+      description: >
+        DatasetPreviewTablesResponse returns information about the table structure of a dataset.
+
+    TableModel:
```

Comment:
I still have questions about this API and what it needs to provide. The API this is modeled after TDR's `/api/repository/v1/snapshots/{id}?include=TABLES`. However the UI doesn't need many of the fields that are in this response. I believe the only data needed by the UI is:
- the table names
- for each table, the number or rows (since we disable selecting empty tables in the UI)

Column names aren't needed as the column names are included in the preview API that returns the table contents.

I think it would be best to have the smallest API possible that satisfies the UI's needs, to make it easier to add support for different storage systems.